### PR TITLE
Added reduce() function for FindWrapper

### LINKED ISSUE #62: reduce() function for FindWrapper
FindWrapper is missing reduce() function.

### PR DESCRIPTION
**Pull request by DzikowskiW**
Resolves #62

### CODE REVIEW

**Comment by gnarf** [comment]
I really need to see a use case for this...

I'm having a hard time understanding why you'd want a reduce that does FindWrappers of the nodes...  Map is an easy one for me, but reduce is getting a bit out there.

File: src/preact-render-spy.js [hunk 1]
```diff
@@ -270,6 +270,17 @@ class FindWrapper {
     );
   }
 
+  reduce(fn, initialValue) {
+    verifyFoundNodes(this);
+    if (initialValue === undefined) {
```
Comment (mzgoddard) [inline src/preact-render-spy.js]: Use the typeof check.

```js
typeof initialValue === 'undefined'
```

**Comment by DzikowskiW** [comment]
@gnarf there is actually a decent [example in enzyme's documentation](https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/reduce.md). I think `.map()` is much more useful and if you want to keep preact-render-spy's API a bit more constrained, `.reduce()` is not a necessity, but otherwise it helps once in a while. 

If we decide to keep this function, I will add a better README example. Let me know what's your decision on that.

**Comment by gnarf** [comment]
> If we decide to keep this function, I will add a better README example. Let me know what's your decision on that.

I think I need to see that example to want to keep the function :)

**Pull request closed without merging**